src/inmanta/db/versions/v202211230.py

```diff
@@ -0,0 +1,98 @@
+"""
+    Copyright 2022 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+from asyncpg import Connection
+
+
+async def update(connection: Connection) -> None:
+    schema = f"""
+    ALTER TABLE ONLY public.resource
+        DROP CONSTRAINT resource_pkey;
+
+    ALTER TABLE ONLY public.resource
+        ADD PRIMARY KEY (environment, model, resource_id);
```

Comment:
good question, what is you opinion after the training?

Reply:
The existing index is: `CREATE UNIQUE INDEX resource_env_resourceid_index ON resource (environment, resource_id, model DESC);`. I did some research from curiosity and it seems that you cannot define the order (ASC/DESC) using a primary key but you can when you define an index manually. However, it's possible to indicate that a primary key should use a certain index. So we should be able to do something like this: `ALTER TABLE ONLY public.resource ADD PRIMARY KEY USING INDEX resource_env_resourceid_index`.

Reply:
Interesting, I didn't know that.

To answer Wouter's question, we didn't really go in-depth on this, but as far as I can tell for the purposes of the primary key using the existing index should suffice, and for the purposes of index performance it should give the same behavior that we had before this PR. That said, I can't say for sure if there might be any queries that might benefit more from this ordering. Since we don't have a specific query in mind that would, I would be inclined to just use the existing index the way Arnaud commented above.